test/decimal.test.js

```diff
@@ -8,59 +8,195 @@
+      OrderDecimal.find({where: {count: '0.0005'}}, function(err, orders) {
+        if (err) return done(err);
+        orders.length.should.be.above(0);
+        const o = orders[0];
+        o.count.should.deepEqual(Decimal128.fromString('0.0005'));
+        o.count.should.be.an.instanceOf(Decimal128);
+        done();
+      });
+    });
 
-  it('create - coerces strings to decimal 128', function(done) {
-    OrderDecimal.create({count: '0.0005'}, function(err, order) {
-      if (err) return done(err);
-      // FIXME Ideally the returned data should have a decimal `count`
-      // not a string. While juggler generates the returned data, connector
-      // doesn't have any control.
-      order.count.should.equal('0.0005');
-      done();
+    it('destroyAll - deletes all with where', function(done) {
+      OrderDecimal.create({count: '0.0006'})
+        .then(function() {
+          return OrderDecimal.destroyAll({count: '0.0005'});
+        })
+        .then(function() {
+          return OrderDecimal.find();
+        })
+        .then(function(r) {
+          r.length.should.equal(1);
+          r[0].count.should.deepEqual(Decimal128.fromString('0.0006'));
+          done();
+        })
+        .catch(done);
     });
   });
 
-  it('find - filters decimal property', function(done) {
-    OrderDecimal.find({where: {count: '0.0005'}}, function(err, orders) {
-      if (err) return done(err);
-      orders.length.should.be.above(0);
-      const o = orders[0];
-      o.count.should.deepEqual(Decimal128.fromString('0.0005'));
-      o.count.should.be.an.instanceOf(Decimal128);
-      done();
+  describe('model with decimal property - nested array', function() {
+    before(function(done) {
+      db = global.getDataSource();
+      var propertyDef = {
+        // nested property in an array
+        lines: [
+          {
+            unitPrice: {
+              type: 'string',
+              title: 'The unitPrice Schema ',
+              mongodb: {
+                dataType: 'Decimal128',
+              },
+            },
+          },
+        ],
+      };
+
+      OrderDecimalArr = db.createModel('OrderDecimalNested', propertyDef);
+      OrderDecimalArr.destroyAll(done);
+    });
+
+    it('create - coerces strings to decimal 128', function(done) {
+      const sample = {
+        lines: [{unitPrice: '0.0005'}],
+      };
+      OrderDecimalArr.create(sample, function(err, order) {
+        if (err) return done(err);
+        // FIXME Ideally the returned data should have a decimal `count`
+        // not a string. While juggler generates the returned data, connector
+        // doesn't have any control.
+        JSON.parse(JSON.stringify(order.lines)).should.deepEqual(sample.lines);
+        done();
+      });
+    });
+
+    it('find - filters nested array decimal property', function(done) {
+      const cond = {where: {lines: {elemMatch: {unitPrice: Decimal128.fromString('0.0005')}}}};
```

Comment:
Maybe it might be beneficial to use `beforeEach` block to delete all records, and create/find/delete the records you want in each test. This removes hard dependencies between tests and test data pollution.

Reply:
@b-admike There are 3 models: OrderDecimal, OrderDecimalArr, OrderDecimalObj, so the 3 test suites won't have data pollution problem :)

And the 3 `it` tests in each `describe` block should run in order, it's on purpose (and please note I do clean up the data by running `destroyAll` in the before function):
- create a sample
- find the created sample with filter
- create another sample with different data --> destroy all data with filter --> find all data --> only one instance returned.
- find it by query